lib/queue.js

```diff
@@ -420,11 +420,20 @@ class Queue extends Emitter {
     return promise;
   }
 
-  _waitForJob() {
-    const idPromise = helpers.deferred();
+  _waitForJob(testingCb) {
+    let idPromise = helpers.deferred();
     this.bclient.brpoplpush(this.toKey('waiting'), this.toKey('active'), 0,
       idPromise.defer());
 
+    // The only reason for this callback is to support deterministic testing of
+    // removeJob between the time brpoplpush resolves and Job.fromId() runs
```

Comment:
Instead of modifying the source to support the test, what do you think about modifying the test to either stub the `Job.fromId` static method or modify the `brpoplpush` function to provide this hook?

Reply:
Yes, I had thought about wrapping `Job.fromId()` -- I'll revisit that approach.